### ISSUE TITLE
Azure Explorer toolkit not working after update of rider 2020.3 .1

### ISSUE THREAD

**Issue by ghost** [JetBrains/azure-tools-for-intellij#429]
The azure explorer was working till I update the rider to the 2020.3.1 . Buid #RD-203-6682.13'

![image](https://user-images.githubusercontent.com/1625641/103338387-49e88800-4aa4-11eb-8c65-c93321f391ab.png)

I did the SigOut and SignIn, but still facing the same issue.

I can see I am signed In 
![image](https://user-images.githubusercontent.com/1625641/103338491-96cc5e80-4aa4-11eb-9c38-02ba9fbaa39b.png)

But if I select refresh or subscriptions I get the same error as above.

OS - MAC cataline

**Comment by maartenba** [comment]
@anandjaisy How are you signed in, with Azure CLI or device flow?

**Comment by ghost** [comment]
> @anandjaisy How are you signed in, with Azure CLI or device flow?

@maartenba  Device flow

**Comment by maartenba** [comment]
When you sign out and then in again, is a dialog shown to select subscriptions?

Could you try closing Rider, then remove `$HOME\.AzureToolsForIntelliJ` (e.g. `C:\Users\you\.AzureToolsForIntelliJ` on Windows)

**Comment by ghost** [comment]
> When you sign out and then in again, is a dialog shown to select subscriptions?
> 
> Could you try closing Rider, then remove `$HOME\.AzureToolsForIntelliJ` (e.g. `C:\Users\you\.AzureToolsForIntelliJ` on Windows)


No, the dialog doesn't show any subscriptions I get the error over there only. What is the location for AzureToolsForIntelliJ on MAC ?

**Comment by maartenba** [comment]
That would be `$HOME/.AzureToolsForIntelliJ`

**Comment by ghost** [comment]
$HOME/.AzureToolsForIntelliJ this location doesn't seem to be there on my machine.

![image](https://user-images.githubusercontent.com/1625641/103351258-a52c7180-4ac8-11eb-8b09-0397e9b20f65.png)

Doesn't navigate anywhere

**Comment by maartenba** [comment]
Perhaps as `~/.AzureToolsForIntelliJ`?

**Comment by ghost** [comment]
After delete the folder, same issue I am facing.

**Comment by maartenba** [comment]
Would you mind trying with Azure CLI login instead?

**Comment by sdubov** [comment]
@anandjaisy, could you please send us `frontend.log` file (`Help | Diagnostic Tools | Show Log in Finder`)? You can find an extra instruction how to send them to us privately [here](https://rider-support.jetbrains.com/hc/en-us/articles/360001079770-How-to-collect-Rider-logs-).

**Comment by ghost** [comment]
Done, should be in the new support request

**Comment by sdubov** [comment]
Thank you @anandjaisy for sharing. I've checked the log. It shows the issue with gathering valid Azure subscription when you try to login. Let's try to step backward first. Could you please confirm that you could login to web `portal.azure.com` using the same subscription?

Similar reports from origin:
- https://github.com/microsoft/azure-tools-for-java/issues/4790
- https://github.com/microsoft/azure-tools-for-java/issues/4792

**Comment by ghost** [comment]
@sdubov Yes I can, I was able to log in as well before the rider update. After the rider update to the latest version, I am not able to publish to Azure.

![Screen Shot 2020-12-30 at 8 51 25 pm](https://user-images.githubusercontent.com/1625641/103363556-5c34e700-4ae1-11eb-9247-cbf02838a43d.png)

**Comment by maartenba** [comment]
Let's see what those two tickets with Microsoft bring.

In the meantime, could you try Azure CLI?

**Comment by ghost** [comment]
With Azure CLI, it is working fine

**Comment by maartenba** [comment]
Thanks! We'll investigate further, but happy to hear you are unblocked for now.

**Comment by dperber** [comment]
Any update on this?

**Comment by maartenba** [comment]
@dperber Not at the moment. Using Azure CLI is the currently recommended workaround.
